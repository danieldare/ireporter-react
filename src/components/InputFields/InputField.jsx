import React from 'react';
import PropTypes from 'prop-types';
import './inputField.css';

const InputField = ({
  name,
  placeholder,
  value,
  label,
  error,
  type,
  info,
  onChange,
  disabled,
  classname,
  focus,
  accept,
  multiple
}) => {
  return (
    <div>
      <input
        type={type}
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        disabled={disabled}
        className={`input-field ${classname}`}
        onFocus={focus}
        accept={accept}
        multiple={multiple}
      />
      {error && <small className="info-error">{error}</small>}
    </div>
  );
};

export const TextAreaField = ({
  name,
  placeholder,
  value,
  label,
  error,
  info,
  onChange,
  disabled,
  classname,
  focus
}) => {
  return (
    <div>
      <textarea
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        disabled={disabled}
        className={`textarea-field ${classname}`}
        onFocus={focus}
      />
      {error && <small className="info-error">{error}</small>}
    </div>
  );
};

InputField.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.string
};

TextAreaField.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.string
};

InputField.defaultProps = {
  type: 'text'
};

export default InputField;
