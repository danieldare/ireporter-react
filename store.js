import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './src/reducers';

const initialState = {};

const middlewares = [thunk];

if (process.env.NODE_ENV === `development`) {
  middlewares.push(createLogger());
}

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middlewares))
);

export default store;
