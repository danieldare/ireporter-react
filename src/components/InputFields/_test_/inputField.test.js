import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import InputField from '../InputField';
import store from '../../../../store';
import mockRouterOptions from '../../../../test/__mocks__/mockRouter';

describe('<InputField/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <InputField />
        </Router>
      </Provider>,
      mockRouterOptions
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
