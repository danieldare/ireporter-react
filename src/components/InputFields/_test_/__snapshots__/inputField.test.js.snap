// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField/> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <InputField
        type="text"
      >
        <div>
          <input
            className="input-field undefined"
            type="text"
          />
        </div>
      </InputField>
    </Router>
  </BrowserRouter>
</Provider>
`;
